### ISSUE TITLE
Store variables 

### ISSUE THREAD

**Issue by kechenkristin** [jenkins-infra/backend-extension-indexer#82]
### What feature do you want to see added?

Currently, the properties and environment variables are hard-coding inside the codes(i.e SourceAndLibs.java), in the future, it's better to store properties and environment variables separately (i.e.https://commons.apache.org/proper/commons-configuration/apidocs/org/apache/commons/configuration2/PropertiesConfiguration.html)  so that it's easier to maintain and reduce security issues potentially.

**Comment by MarkEWaite** [comment]
I don't think that there is anyone that is willing to review your proposed changes @Gowtham-arja .  I recommend you find issues where there is an active maintainer that is willing to review changes.  I keep dependencies current on this repository, but am not willing to review most other code contributions to this repository.
